applications/OptimizationApplication/custom_utilities/container_variable_data_holder/collective_variable_data_holder.cpp

```diff
@@ -0,0 +1,412 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Suneth Warnakulasuriya
+//
+
+// System includes
+#include <sstream>
+#include <string>
+#include <vector>
+#include <variant>
+
+// Project includes
+
+// Application includes
+
+// Include base h
+#include "collective_variable_data_holder.h"
+
+namespace Kratos {
+
+///@name Kratos Classes
+///@{
+
+CollectiveVariableDataHolder::CollectiveVariableDataHolder(const std::vector<ContainerVariableDataHolderPointerVariantType>& rContainerVariableDataHolderPointersList)
+{
+    for (const auto& p_container_variable_data_holder : rContainerVariableDataHolderPointersList){
+        this->AddVariableDataHolder(p_container_variable_data_holder);
+    }
+}
+
+CollectiveVariableDataHolder::CollectiveVariableDataHolder(const CollectiveVariableDataHolder& rOther)
+{
+    for (const auto& p_container_variable_data_holder : rOther.mContainerVariableDataHolderPointersList) {
+        std::visit([&](auto&& v) {
```

Comment:
```suggestion
        std::visit([&](const auto& v) {
```

Reply:
this applies to all other lambdas you're using in `std::visit` as well.

Reply:
alternatively, you could define an overload that takes a temporary, and then moves it into the correct container.